### ISSUE TITLE
the problem of the score and the feature

### ISSUE THREAD

**Issue by cuixianheng** [alinlab/CSI#9]
1、In your paper ,  the score is related of the cosine similarity and the norm of the feature. But in your code, you don't use the cosine similarity to calculate the score.
2、in your code , features are extracted after the projection head， but the simclr's feature is extracted after the resnet.
Look forward to your reply

**Comment by sangwoomo** [comment]
Hi,

1. We use the [inner product](https://github.com/alinlab/CSI/blob/97ed26ba5f005ffbc5dba91c378d288e76bf2b88/evals/ood_pre.py#L120) in our implementation, which is the product of cosine similarity and norm.

2. We found that the feature *after projection* works better for detecting OOD samples; hence, we use it except for linear evaluation (feature *before projection* is better for linear evaluation, as suggested in the SimCLR paper). We mentioned it on page 3 of our paper, but sorry that it was not super clear. We remark that we tested various detection scores on both the feature before and after projection, but our proposed CSI score was the best among them.
